pkgs/by-name/gl/glfw3/package.nix

```diff
@@ -26,7 +26,7 @@ stdenv.mkDerivation {
     ./0009-Defer-setting-cursor-position-until-the-cursor-is-lo.patch
   ];
 
-  propagatedBuildInputs = [ libGL ];
+  propagatedBuildInputs = lib.optionals (!stdenv.hostPlatform.isWindows) [ libGL ];
```

Comment:
Alternative: I could do `!libGL.meta.unsupported` but that's like more implicit?

Reply:
The way you've done it is preferred because its more clear. The latter would require knowing what libGL considers unsupported

Reply:
Indeed, but the functional difference is that there could be more platforms that do not need/use libGL that would be unblocked by using `meta.unsupported`. Yet another option to consider is: `isUnix`

Reply:
`meta.unsupported` isn't good to use with cross because it only uses one one platform
```
nix-repl> pkgsCross.mingwW64.pkgsHostTarget.libGL.meta.unsupported
true

nix-repl> pkgsCross.mingwW64.pkgsBuildHost.libGL.meta.unsupported
false
```

Instead use `lib.meta.availableOn stdenv.hostPlatform libGL`

```
nix-repl> lib.meta.availableOn pkgsCross.mingwW64.stdenv.hostPlatform libGL
false
```

Reply:
```suggestion
  propagatedBuildInputs = lib.optionals (lib.meta.availableOn stdenv.hostPlatform libGL) [ libGL ];
```

Reply:
@Artturin indeed, I was meaning to suggest `meta.unsupported` as a shorthand to `availableOn`. The downside of this approach, on the other hand, which I'm referring to in https://github.com/NixOS/nixpkgs/pull/342271#discussion_r1760938071 and how I understand https://github.com/NixOS/nixpkgs/pull/342271#discussion_r1760992136 is that if somebody were to change the `platforms` of `libGL`, `imgui`'s and `glfw`'s `platform` would change silently.

With `!isWindows` the error would be: `libglvnd-... not available on requested platforms`.
With `availableOn` the error is: `imgui-... not available on requested platforms`.

IMO we really want the former:)